### ISSUE TITLE
Envoy Gateway Support 

### ISSUE THREAD

**Issue by alexsnaps** [Kuadrant/kuadrant-operator#325]
While there isn't first class support for [Wasm](https://github.com/envoyproxy/gateway/issues/2025) extensions yet in Envoy Gateway, it might be feasible to use [`EnvoyPatchPolicy`](https://gateway.envoyproxy.io/latest/design/envoy-patch-policy/) to achieve some of our goals right now, possibly with the use of the [Extension Manager](https://gateway.envoyproxy.io/latest/design/extending-envoy-gateway/#registering-extensions-in-envoy-gateway). 

This issue's goal is to track the work and/or dependencies so we can support Envoy Gateway as Gateway API provider.

## Initial Tasks

 - [x] #567
 - [x] Spike into seeing what's missing to add support for _Policy Attachment_ to Envoy Gateway, just like it does for `ExtensionFilter`s.
 - [x]  #645 
 - [x]  #681
 - [x] #686

## Tasks
- [x] #678
- [ ] Rate limit policy implementation: Rate Limit EnvoyExtensionPolicy controller
- [ ] Authorization policy implementation: SecurityPolicy controller

## Further questions

 - [x] How would [EG calling back into the Kuadrant Operator](https://gateway.envoyproxy.io/v0.6.0/design/extending-envoy-gateway/) about topology changes be addressed and what would it mean for our [DAG proposal](https://github.com/Kuadrant/architecture/issues/29)? 
 - [x] Are there issues in having out-of-order changes handled by EG and Kuadrant mutations of the config thru [`EnvoyPatchPolicy`](https://gateway.envoyproxy.io/latest/design/envoy-patch-policy/) 
 - [x] Are there unresolved dependencies in EG for us to get this done at all?

**Comment by eguzki** [comment]
Added https://github.com/eguzki/poc-kuadrant-envoy-gateway/issues/2 to report findings

**Comment by eguzki** [comment]
> How would [EG calling back into the Kuadrant Operator](https://gateway.envoyproxy.io/v0.6.0/design/extending-envoy-gateway/) about topology changes be addressed and what would it mean for our https://github.com/Kuadrant/architecture/issues/29?

That's the [Envoy Extension Manager](https://gateway.envoyproxy.io/contributions/design/extending-envoy-gateway/) and EnvoyGateway has already APIs (CRDs) to do the same. No need to implement service endpoint to patch envoy xDS resources. Actually, the clean way to go for WASM is the [EnvoyExtensionPolicy](https://gateway.envoyproxy.io/contributions/design/envoy-extension-policy/) where WASM will eventually be supported. It is just not there yet. This https://github.com/envoyproxy/gateway/pull/3164 already merged PR needs to be shipped in a release. Hopefully in v1.1 

> Are there issues in having out-of-order changes handled by EG and Kuadrant mutations of the config thru [EnvoyPatchPolicy](https://gateway.envoyproxy.io/latest/design/envoy-patch-policy/)

TL;DR: we need to investigate when EnvoyExtensionPolicy API is there available. WE need to enforce that rate limiting happens after external auth filter. 

> Are there unresolved dependencies in EG for us to get this done at all?

[EnvoyExtensionPolicy](https://gateway.envoyproxy.io/contributions/design/envoy-extension-policy/) API. But we can start using it by patching configuration in raw mode using EnvoyPathPolicy, which requires advanced users who are well versed in Envoy xDS API and its capabilities. Furthermore, the API might remain unstable across versions due to potential changes in the Envoy Proxy API or Envoy Gateway’s xDS translation
